service/src/main.rs

```diff
@@ -443,6 +455,50 @@ fn start_interval_trusted_getter_execution<E: Sidechain>(enclave_api: &E) {
 	);
 }
 
+/// Send extrinsic to chain according to the market data update interval in the settings
+/// with the current market data (for now only exchange rate).
+fn start_interval_market_update<E: EnclaveBase + TeeracleApi>(
+	api: &Api<sr25519::Pair, WsRpcClient>,
+	enclave_api: &E,
+) {
+	use itp_settings::node::MARKET_DATA_UPDATE_INTERVAL;
+
+	schedule_on_repeating_intervals(
+		|| {
+			execute_update_market(api, enclave_api);
+		},
+		MARKET_DATA_UPDATE_INTERVAL,
+	);
+}
+
+fn execute_update_market<E: EnclaveBase + TeeracleApi>(
+	node_api: &Api<sr25519::Pair, WsRpcClient>,
+	enclave: &E,
+) {
+	// Get market data for usd (hardcoded)
+	let uxt = match enclave.update_market_data_xt("usd") {
```

Comment:
an exchange rate always has a trading pair. Just naming USD is meaningless. it should rather be `DOT/USD`

Reply:
Can we choose any cryptocurrency? Is it not the market data related to the chain token? In my implementation, I assume this.